### PR TITLE
Fix infinite loop in InstrumentPlayHandle::play

### PR DESCRIPTION
**Pull request by sakertooth**
Regression from #7143. `nphsLeft` was not being set to `false` after each iteration, meaning that it was left being equal to `true` after the first `NotePlayHandle` was found.

~This variable was removed completely. A simple for loop that busy waits after the call to `ThreadableJob::process` is now whats being done.~ (This prevented multiple handles from being queued at once)

### CODE REVIEW

File: src/core/InstrumentPlayHandle.cpp [hunk 1]
```diff
@@ -47,22 +47,20 @@ void InstrumentPlayHandle::play(sampleFrame * working_buffer)
...
 		{
-			if (constNotePlayHandle->state() != ThreadableJob::ProcessingState::Done &&
-				!constNotePlayHandle->isFinished())
-			{
-				nphsLeft = true;
-				NotePlayHandle * notePlayHandle = const_cast<NotePlayHandle *>(constNotePlayHandle);
-				notePlayHandle->process();
-			}
+			const_cast<NotePlayHandle*>(constNotePlayHandle)->process();
+			while (notFinished()) {}
```
Comment (Veratil) [inline src/core/InstrumentPlayHandle.cpp:60]: I feel like this fundamentally changes the way it's processing from before.

Before:
Loop through all handles, if one is not done call it to process in its thread. Loop until done.

After:
Loop through all handles, if one is not done call it to process and spin wait until that single thread is done. Continue to the next handle.

Comment (Veratil) [reply src/core/InstrumentPlayHandle.cpp:60]: Oh you said in the description that's on purpose. 😁

Comment (sakertooth) [reply src/core/InstrumentPlayHandle.cpp:60]: The previous behavior where it looped over all the handles multiple times didn't sit right with me. It felt inefficient, assuming we have a bit of handles in there.

Comment (Veratil) [reply src/core/InstrumentPlayHandle.cpp:60]: Before all the threads can awake and process. Now they are handled one by one only.

Comment (sakertooth) [reply src/core/InstrumentPlayHandle.cpp:60]: Oh, I see now. I'll change it back.

**Review by Veratil — APPROVED** [review]

**Pull request merged**